### PR TITLE
Replace multiple "or"-seperated startswith and endswith occurences

### PR DESCRIPTION
**Pull request by MSeifert04**
This is a refactoring of the code (it doesn't yield any speedups, nor should it slow down the code, without any change in behaviour) inspired by https://github.com/astropy/astropy/pull/5312#discussion_r78013543.

### CODE REVIEW

File: astropy/table/meta.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def _get_col_attributes(col):
     attrs['name'] = col.info.name
 
     type_name = col.info.dtype.type.__name__
-    if six.PY3 and (type_name.startswith('bytes') or type_name.startswith('str')):
+    if six.PY3 and type_name.startswith(('bytes', 'str')):
```
Comment (pllim) [inline astropy/table/meta.py]: This will conflict with #5312. Should we wait till that one is merged first and then rebase this one?

**Comment by pllim** [comment]
If nothing else, it makes the codes easier to read, so :+1: !

File: astropy/table/meta.py (hunk 1 above)
Comment (MSeifert04) [inline astropy/table/meta.py]: I can wait until the other one is merged and then rebase this one.

**Comment by mhvk** [comment]
Like this too, but as @pllim suspected, merging #5312 causes a merge conflict.  Happy to put this in once that is resolved...

**Comment by MSeifert04** [comment]
it is rebased.

**Comment by mhvk** [comment]
OK, great, so merging! Thanks!

**Pull request merged**